tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/ContactDetailsValidator.java

```diff
@@ -51,12 +58,16 @@ public void validate(ContactDetailsDTO dto) throws MethodArgumentNotValidExcepti
     final boolean currentOnly = false;
     List<FieldError> fieldErrors = new ArrayList<>();
     fieldErrors.addAll(checkTitle(dto, currentOnly));
+    fieldErrors.addAll(checkEmail(dto));
 
     if (!fieldErrors.isEmpty()) {
       BeanPropertyBindingResult bindingResult =
           new BeanPropertyBindingResult(dto, CONTACT_DETAILS_DTO_NAME);
       fieldErrors.forEach(bindingResult::addError);
-      throw new MethodArgumentNotValidException(null, bindingResult);
+
+      Method method = this.getClass().getMethods()[0];
```

Comment:
Isn't this a little brittle? I'm not an expert with using reflection, but it seems a bit of a gotcha to reference the ContactDetailsValidator through its position in the public methods array. I could be wrong, so please disregard if I've missed something.

Reply:
For example, I might have thought this.getClass().getMethod("ContactDetailsValidator") might have been a bit clearer?

Reply:
I found if we set the first parameter of `MethodArgumentNotValidException` to null, the `getMessage()` of the Exception class will throw an NPE in the console. So I copy-pasted those reflection codes from another validator.
Thank you for raising this problem. I think the method used here should be the current `validate` method (the 2nd one). Besides, `getMethod` throws an NoSuchMethodException, I have nothing to deal with it, so just throw it to the framework.

Reply:
Works for me! :)